frontend/src/services/inviteService.ts

```diff
@@ -1,26 +1,127 @@
+   *
+   * @typedef {object} Invite
+   * @property {string} email - invitee's email
+   * @property {string} token - invite token
+   *
+   * @returns {Array<Invite>} array of invite email / token pairs
+   */
+  async createInvites(
+    resourceType: string,
+    resource: any,
+    currentUser: any,
+    emails: Array<string>,
+    expiresAt?: number,
+    permCodes?: Array<string>
+  ) {
+
+    const inviteData  ={
+      bucketId: resourceType === 'bucket' ? resource?.bucketId : undefined,
+      objectId: resourceType === 'object' ? resource.id : undefined,
+      expiresAt: expiresAt,
+      permCodes: permCodes
+    };
+
+    // if emails param exists
+    if(emails && emails.length > 0){
+      // create COMS invites
+      const invites = await Promise.all(
+        emails.map(async e => {
+          const token = (await comsAxios().post(`${PATH}`, {
+            ...inviteData,
+            email: e,
+          })).data;
+          return { email: e, token: token};
+        })
+      );
+      // send invite email notifications
+      await this.emailInvites(resourceType, resource, currentUser, invites);
+      return invites;
+    }
+    // else no emails provided so make an 'open' invite
+    else {
+      const token = (await comsAxios().post(`${PATH}`, inviteData)).data;
+      return [{ token: token }];
+    }
+  },
+
   /**
-   * @function createInvite
-   * Post an Invite, exclusive or on bucketId and objectId
-   * @param {string} bucketId
-   * @param {string} objectId
-   * @param {string} email to be used for access
-   * @param {string} expiration timestamp for token
-   * @returns {string} uuid token of the invite
-   * @returns {array} permCodes token of the invite
+   * @function emailInvites
+   * Semd email to each invitee containing a link to the resource
+   * each email needs a unique invite link url so use CHES merge feature
+   * ref: https://ches.api.gov.bc.ca/api/v1/docs#tag/EmailMerge/operation/postMerge   *
+   * @param {string} resourceType eg bucket or object
+   * @param {COMSObject | Bucket } resource COMS object or bucket
+   * @param {User | null} currentUser current user creating the invite
+   * @param {Array<string>} invites array of email adddresses
+   * @returns {Promise<string>} CHES TransactionId
    */
-  createInvite(bucketId?: string, email?: string, expiresAt?: number, objectId?: string, permCodes?: Array<string>) {
-    return comsAxios().post(`${PATH}`, {
-      bucketId: bucketId || undefined,
-      email: email || undefined,
-      expiresAt: expiresAt || undefined,
-      objectId: objectId || undefined,
-      permCodes: permCodes || undefined
-    });
+  emailInvites(resourceType: string, resource: any, currentUser: any, invites: any){
+    try {
+      // build template
+      let resourceName, subject, body;
+      const currentUserEmail = `<a href="mailto:${currentUser.email}" style="color: #1a5a96 !important">${currentUser.email}</a>`;
```

Comment:
I guess this works, but surely we could separate out all the templating and actual HTML into a separate file?